lib/ast-utils/docs.js

```diff
@@ -319,41 +319,51 @@ function cleanProperty(property) {
 /**
  * @param {string} elementId
  * @param {DocumentAST} domModule
+ * @param {DocumentAST} scriptElement The script that the element was defined in.
  */
-function _findElementDocs(elementId, domModule) {
-  if (!domModule) {
-    return null;
-  }
+function _findElementDocs(elementId, domModule, scriptElement) {
   // Note that we concatenate docs from all sources if we find them.
   var found = [];
 
-  // Do we have a HTML comment on the `<dom-module>`?
+  // Do we have a HTML comment on the `<dom-module>` or `<script>`?
   //
   // Confusingly, with our current style, the comment will be attached to
   // `<head>`, rather than being a sibling to the `<dom-module>`
```

Comment:
Due to the rules around where HTML comments end up in the AST, we have to go on a wild goose chase to find element docs. They can be: 
1) A sibling to the dom-module and part of body
2) The last child of head
2) the last child of the document